pandas/tests/indexes/test_indexing.py

```diff
@@ -148,3 +150,51 @@ def test_getitem_deprecated_float(idx):
 
     expected = idx[1]
     assert result == expected
+
+
+@pytest.mark.parametrize(
+    "idx, target, expected",
+    [
+        ([np.nan, "var1", np.nan], [np.nan], np.array([0, 2], dtype=np.intp)),
+        (
+            [np.nan, "var1", np.nan],
+            [np.nan, "var1"],
+            np.array([0, 2, 1], dtype=np.intp),
+        ),
+        (
+            np.array([np.nan, "var1", np.nan], dtype=object),
+            [np.nan],
+            np.array([0, 2], dtype=np.intp),
+        ),
+        (
+            DatetimeIndex(["2020-08-05", NaT, NaT]),
+            [NaT],
+            np.array([1, 2], dtype=np.intp),
+        ),
+        (
+            np.array([np.nan, -1, np.nan], dtype=object),
+            [np.nan],
+            np.array([0, 2], dtype=np.intp),
+        ),
+        (
+            np.array([np.nan, -1, np.nan], dtype=object),
+            [-1],
+            np.array([1], dtype=np.intp),
+        ),
+        (
+            np.array([np.nan, -1, np.nan], dtype=object),
+            [np.nan, -1],
+            np.array([0, 2, 1], dtype=np.intp),
+        ),
+        (
+            [float("NaN"), "var1", float("NaN")],
+            [float("NaN")],
+            np.array([0, 2], dtype=np.intp),
+        ),
+    ],
+)
+def test_get_indexer_non_unique_multiple_nans(idx, target, expected):
+    # GH 35392
+    axis = Index(idx)
```

Comment:
total nit but i'd combine these two lines into 
``` python
result = Index(axis).get_indexer_for(target)
```

Reply:
agreed we can combine - I'll change